### PR TITLE
Fixes on the dotnet end-to-end test scripts to get it running on linux

### PR DESCRIPTION
**Pull request by shahasad**

### CODE REVIEW

**Comment by shahasad** [comment]
>           Please search 'ca-certificates' in this file, replace the second one with 'apt-transport-https'.

which file? install_ubuntu,sh? shouldn't that ca-certificate change be a separate change?

**Comment by snnn** [comment]
> > ```
> >       Please search 'ca-certificates' in this file, replace the second one with 'apt-transport-https'.
> > ```
> 
> which file? install_ubuntu,sh? shouldn't that ca-certificate change be a separate change?

Yes. See #320
Indeed it was a mistake made by Raymond. He said you will fix it in this PR.

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh [hunk 1]
```diff
@@ -16,14 +19,25 @@ docker build -t "onnxruntime-$IMAGE" --build-arg OS_VERSION=16.04 --build-arg PY
...
         --rm \
         --name "onnxruntime-cpu" \
         --volume "$SOURCE_ROOT:/onnxruntime_src" \
         --volume "$BUILD_DIR:/home/onnxruntimedev" \
-        --volume "$HOME/.cache/onnxruntime:/root/.cache/onnxruntime" \
+        --volume "$HOME/.cache/onnxruntime:/home/onnxruntimedev/.cache/onnxruntime" \
         "onnxruntime-$IMAGE" \
         /bin/bash /onnxruntime_src/csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh \
-        /home/onnxruntimedev/$NUGET_REPO_DIRNAME /onnxruntime_src &
+        /home/onnxruntimedev/$NUGET_REPO_DIRNAME /onnxruntime_src /home/onnxruntimedev $TestDataUrl $TestDataChecksum &
```
Comment (jignparm) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: Are we setting TestDataUrl outside, or is it another positional argument on the command line?

Comment (shahasad) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: it is set as a build environment variable

**Comment by snnn** [comment]
> > ```
> >       Please search 'ca-certificates' in this file, replace the second one with 'apt-transport-https'.
> > ```
> 
> which file? install_ubuntu,sh? shouldn't that ca-certificate change be a separate change?

Yes.  Indeed it was a mistake made by Raymond. He said you will fix it in this PR.

It should be 
```
        git apt-transport-https \
        ca-certificates \
```

Instead of:

```
        git ca-certificates \
        ca-certificates \
```

**Comment by shahasad** [comment]
OK. I'll do it. But this line was last changed 2 months ago, so I don't know it's meaning.

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh [hunk 2]
```diff
@@ -4,26 +4,42 @@
...
 LocalNuGetRepo=$1
 SourceRoot=$2
+BuildDir=$3
+TestDataUrl=$4
+TestDataChecksum=$5
 
-@echo "Downloading test data"
-python build.by --update --download_test_data
+#TestDataUrl=https://onnxruntimetestdata.blob.core.windows.net/models/20190107.zip
+#TestDataChecksum=67622404e651317e24c581350718331a
```
Comment (jignparm) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest.sh]: should remove

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh (hunk 1 above)
Comment (jignparm) [reply csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: this script should be usable from command line as well, outside the build. It's the primary script for building on Linux.

File: csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh [hunk 3]
```diff
@@ -6,8 +6,11 @@
 
 #TODO: Get this working, not tested yet
 
+
```
Comment (jignparm) [inline csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests/runtest-docker.sh]: "set -x -e -o" for helpful debugging info.

**Pull request merged**